src/Materials/Textures/Packer/packer.ts

```
@@ -0,0 +1,783 @@
+                }
+            success(pack);
+            } catch (err) {
+                return;
+            }
+        });
+
+        downloadPromise.then((result) => {
+            let data = "data:text/json;charset=utf-8," + encodeURIComponent(JSON.stringify(result, null, 4));
+            let _a = document.createElement('a');
+            _a.setAttribute("href", data);
+            _a.setAttribute("download", this.name + "_texurePackage.json");
+            document.body.appendChild(_a);
+            _a.click();
+            _a.remove();
+        });
+    }
+
+    /**
+    * Public method to load a texturePacker JSON file.
+    * @param data of the JSON file in string format.
+    */
+    public updateFromJSON(data: string): void {
+            try {
+                let parsedData: ITexturePackerJSON = JSON.parse(data);
+                this.name = parsedData.name;
+                let _options = Object.keys(parsedData.options);
+
+                for (let i = 0; i < _options.length; i++) {
+                    (this.options as any)[_options[i]] = (parsedData.options as any)[_options[i]];
+                }
+
+                for (let i = 0; i < parsedData.frames.length; i += 4) {
+                    let frame: TexturePackerFrame = new TexturePackerFrame(
+                        i / 4,
+                        new Vector2(parsedData.frames[i], parsedData.frames[i + 1]),
+                        new Vector2(parsedData.frames[i + 2], parsedData.frames[i + 3])
+                    );
+                this.frames.push(frame);
+                }
+
+                let channels = Object.keys(parsedData.sets);
+
+                for (let i = 0; i < channels.length; i++) {
+                    let _t = new Texture(parsedData.sets[channels[i]], this.scene, false, false);
+                    (this.sets as any)[channels[i]] = _t;
+                }
+            } catch (e) {

```

we should not swallow errors

---

so return e?

---

nope just not catch

---

or do I use the logger?